### PR TITLE
New Test Suite!

### PR DESCRIPTION
**Pull request by jonnywilliamson**

### CODE REVIEW

**Comment by jonnywilliamson** [comment]
Woah. That was some effort to track down those bugs.

Please merge if you like :)

**Comment by irazasyed** [comment]
Yes, sir! Thanks, Highly appreciated 👍 😄

File: src/Traits/Http.php [hunk 1]
```diff
@@ -227,7 +227,11 @@ protected function replyMarkupToString(array $params): array
     protected function uploadFile(string $endpoint, array $params, $inputFileField): TelegramResponse
     {
         //Check if the field in the $params array (that is being used to send the relative file), is a file id.
-        if (!isset($params[$inputFileField]) || $this->hasFileId($inputFileField, $params)) {
+        if (!isset($params[$inputFileField])) {
```
Comment (jonnywilliamson) [inline src/Traits/Http.php]: This is broken.

Comment (irazasyed) [reply src/Traits/Http.php]: I didn't test since this PR was merged. Please create a PR with a fix if you found the bug? Thanks.

**Pull request merged**